File: FrontEnd/lib/domain/dtos/chat/send_message_request_dto.dart
/// @file send_message_request_dto.dart
/// @description DTO de envío de mensaje en una conversación de chat.
/// El conversationId va en la ruta URL, no en el body.
/// @module Chat
/// @layer Domain
library;

// ═══════════════════════════════════════════════════════════════════════════════
// SEND MESSAGE REQUEST DTO
// ═══════════════════════════════════════════════════════════════════════════════

/// DTO que representa el body de una petición de envío de mensaje.
class SendMessageRequestDto {
  /// Contenido textual del mensaje.
  final String  content;

  /// ID temporal del cliente para reconciliación optimista (opcional).
  final String? tempId;

  const SendMessageRequestDto({required this.content, this.tempId});

  /// Serializa el DTO a JSON para la petición HTTP.
  Map<String, dynamic> toJson() => <String, dynamic>{
    'text':   content,
    'tempId': tempId,
  }..removeWhere((_, v) => v == null);
}

File: BackEnd/src/domain/dtos/chat/send-message-request.dto.ts
/**
 * @file send-message-request.dto.ts
 * @description DTO de envío de mensaje en una conversación.
 * El conversationId viene del parámetro de ruta URL (no se incluye en el body).
 * @module Chat
 * @layer Domain
 */

import { IsString, IsOptional, MaxLength, IsObject } from 'class-validator';

/**
 * Datos del body para enviar un mensaje en una conversación existente.
 */
export class SendMessageRequestDto {
  /** Texto del mensaje (opcional si se adjunta contenido multimedia) */
  @IsOptional()
  @IsString()
  @MaxLength(2000)
  text?: string;

  /** Metadatos de contenido multimedia adjunto */
  @IsOptional()
  @IsObject()
  contentMeta?: Record<string, unknown>;

  /** ID temporal asignado por el cliente para actualizaciones optimistas */
  @IsOptional()
  @IsString()
  tempId?: string;
}
